File: HealthMonitor/app/src/main/java/com/example/healthmonitor/HealthHistoryAdapter.java
package com.example.healthmonitor;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import java.util.List;

public class HealthHistoryAdapter extends RecyclerView.Adapter<HealthHistoryAdapter.ViewHolder> {

    private List<HealthRecord> healthRecords;

    public HealthHistoryAdapter(List<HealthRecord> healthRecords) {
        this.healthRecords = healthRecords;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_health_record, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        HealthRecord record = healthRecords.get(position);
        holder.tvTime.setText("Time: " + record.getTime());
        holder.tvSpO2.setText("SpO2: " + record.getSpO2() + "%");
        holder.tvHeartRate.setText("Heart Rate: " + record.getHeartRate() + " BPM");
        holder.tvTemperature.setText("Temperature: " + record.getTemperature() + "°C");
    }

    @Override
    public int getItemCount() {
        return healthRecords.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvTime, tvSpO2, tvHeartRate, tvTemperature;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            tvTime = itemView.findViewById(R.id.tvTime);
            tvSpO2 = itemView.findViewById(R.id.tvSpO2);
            tvHeartRate = itemView.findViewById(R.id.tvHeartRate);
            tvTemperature = itemView.findViewById(R.id.tvTemperature);
        }
    }
}

File: HealthMonitor/app/src/main/java/com/example/healthmonitor/adapter/HealthHistoryAdapter.kt
package com.example.healthmonitor.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageButton
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.healthmonitor.R
import com.example.healthmonitor.model.HealthHistory

class HealthHistoryAdapter(
    private var healthHistories: List<HealthHistory>,
    private val onDeleteClick: (HealthHistory) -> Unit
) : RecyclerView.Adapter<HealthHistoryAdapter.ViewHolder>() {

    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val tvDateTime: TextView = view.findViewById(R.id.tvDateTime)
        val tvHeartRate: TextView = view.findViewById(R.id.tvHeartRate)
        val tvSpO2: TextView = view.findViewById(R.id.tvSpO2)
        val tvTemperature: TextView = view.findViewById(R.id.tvTemperature)
        val btnDelete: ImageButton = view.findViewById(R.id.btnDelete)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_health_history, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val history = healthHistories[position]
        holder.tvDateTime.text = history.getFormattedDate()
        holder.tvHeartRate.text = "${history.heartRate} BPM"
        holder.tvSpO2.text = "${history.spO2}%"
        holder.tvTemperature.text = String.format("%.1f°C", history.temperature)
        
        holder.btnDelete.setOnClickListener {
            onDeleteClick(history)
        }
    }

    override fun getItemCount() = healthHistories.size

    fun updateData(newHistories: List<HealthHistory>) {
        healthHistories = newHistories
        notifyDataSetChanged()
    }
} 